contracts/rbn-staking/FeeCustody.sol

```diff
@@ -0,0 +1,425 @@
+      IERC20 asset = IERC20(assets[i]);
+      uint256 assetBalance = asset.balanceOf(address(this));
+
+      if (assetBalance == 0) {
+        continue;
+      }
+
+      uint256 multiSigRevenue = assetBalance
+        .mul(TOTAL_PCT.sub(pctAllocationForRBNLockers))
+        .div(TOTAL_PCT);
+
+      // If we are holding the distributionToken itself,
+      // do not swap
+      if (address(asset) != address(distributionToken)) {
+        // Calculate RBN allocation amount to swap for distributionToken
+        uint256 amountIn = assetBalance.sub(multiSigRevenue);
+        _swap(address(asset), amountIn, _minAmountOut[i]);
+      }
+
+      // Transfer multisig allocation of protocol revenue to multisig
+      asset.safeTransfer(protocolRevenueRecipient, multiSigRevenue);
+    }
+
+    toDistribute = distributionToken.balanceOf(address(this));
+    distributionToken.safeApprove(address(feeDistributor), 0);
+    distributionToken.safeApprove(address(feeDistributor), toDistribute);
+
+    // Tranfer RBN locker allocation of protocol revenue to fee distributor
+    feeDistributor.burn(address(distributionToken), toDistribute);
+  }
+
+  /**
+   * @notice
+   * Amount of _asset allocated to RBN lockers from current balance
+   * @return amount allocated to RBN lockers
+   */
+  function claimableByRBNLockersOfAsset(address _asset)
+    external
+    view
+    returns (uint256)
+  {
+    uint256 allocPCT = pctAllocationForRBNLockers;
+    uint256 balance = _asset == address(0)
+      ? address(this).balance
+      : IERC20(_asset).balanceOf(address(this));
+    return balance.mul(allocPCT).div(TOTAL_PCT);
+  }
+
+  /**
+   * @notice
+   * Amount of _asset allocated to multisig from current balance
+   * @return amount allocated to multisig
+   */
+  function claimableByProtocolOfAsset(address _asset)
+    external
+    view
+    returns (uint256)
+  {
+    uint256 allocPCT = TOTAL_PCT.sub(pctAllocationForRBNLockers);
+    uint256 balance = _asset == address(0)
+      ? address(this).balance
+      : IERC20(_asset).balanceOf(address(this));
+    return balance.mul(allocPCT).div(TOTAL_PCT);
+  }
+
+  /**
+   * @notice
+   * Total allocated to RBN lockers across all assets balances
+   * @return total allocated (in USD)
+   */
+  function totalClaimableByRBNLockersInUSD() external view returns (uint256) {
+    uint256 allocPCT = pctAllocationForRBNLockers;
+    return _getSwapQuote(allocPCT);
+  }
+
+  /**
+   * @notice
+   * Total allocated to multisig across all assets balances
+   * @return total allocated (in USD)
+   */
+  function totalClaimableByProtocolInUSD() external view returns (uint256) {
+    uint256 allocPCT = TOTAL_PCT.sub(pctAllocationForRBNLockers);
+    return _getSwapQuote(allocPCT);
+  }
+
+  /**
+   * @notice
+   * Total claimable across all asset balances based on allocation PCT
+   * @param _allocPCT allocation percentage
+   * @return claimable total claimable (in USD)
+   */
+  function _getSwapQuote(uint256 _allocPCT)
```

Comment:
My feedback here is that `_getSwapQuote` isn't an accurate function name, because it doesn't actually look up the price from Uniswap, but looks up Chainlink. I suggest we go for `_getTotalAssetValue`